src/modules/fetchPage.ts

```diff
@@ -9,7 +9,7 @@ export function fetchPage(this: Swup, data: TransitionOptions): Promise<PageReco
 
 	if (this.cache.exists(url)) {
 		this.triggerEvent('pageRetrievedFromCache');
-		return Promise.resolve(this.cache.getPage(url));
+		return Promise.resolve(this.cache.getPage(url) as PageRecord);
```

Comment:
While the casting is fine thanks to the `exists` check above, I think it would be cleaner to retrieve the record and be working with the return value directly, since it can be `null`.

Reply:
You mean without the `this.cache.exists` and instead with `this.cache.getPage`?

Reply:
Yup! At that point, we already know it's `PageRecord` or `null` and can use that as a check instead of `exists` and don't access the cache once more.